### PR TITLE
[NKV-SDK-Host] Refactor nkv host code

### PR DESCRIPTION
**Pull request by ramachaitanyak**
Brief Changes:
    ==============
    * Fixed issue using resize and pushback at the same time
    * Fix listing_keys and data_cache proto-types
    * Gitlab upgrade, trigger:include now takes a string or list
    * Fix Sonar-Scanner download.
        - Get sonar-scanner from URL determined by CI/CD var
        - Determine sonar-scanner dir name by listing contents of zip

    Testing Done:
    =============
    * Run with gitlab pipeline

    * {Ticket/Issue Reference}
            * SVK-2328

### CODE REVIEW

File: host/src/include_private/nkv_framework.h [hunk 1]
```diff
@@ -282,16 +286,28 @@
...
+          listing_keys[i] = std::unordered_map<std::size_t, std::set<std::string>>();
+      }
       if (nkv_in_memory_exec) {
-        data_cache = new std::unordered_map<std::string, nkv_value_wrapper*> (nkv_listing_cache_num_shards);
+        //data_cache = new std::unordered_map<std::string, nkv_value_wrapper*> (nkv_listing_cache_num_shards);
+        // Resize and initialize data_cache
+        data_cache.resize(nkv_listing_cache_num_shards);
+        for (auto i=0; i<nkv_listing_cache_num_shards; i++) {
+            // Initialize map at index
+            data_cache[i] = std::unordered_map<std::string, nkv_value_wrapper*>();
```
Comment (somnathr) [inline host/src/include_private/nkv_framework.h:297]: How nkv_value_wrapper* will be deleted ?

File: host/src/nkv_framework.cpp [hunk 2]
```diff
@@ -2224,10 +2224,10 @@ NKVTargetPath::~NKVTargetPath() {
...
 
       delete[] cache_rw_lock_list;
-      delete[] listing_keys;
-      if (nkv_in_memory_exec) {
-        delete[] data_cache;
-      }
+      // This delete is not required with the STL container, clean up happens
+      // when container goes out of scope
+      //delete[] listing_keys;
+      //delete[] data_cache;
```
Comment (somnathr) [inline host/src/nkv_framework.cpp]: Please remove commented code

File: .gitlab-ci.yml
Comment (benixon) [inline .gitlab-ci.yml:1]: Why are CI/CD files being modified?

File: .gitlab/sonar.yml
Comment (benixon) [inline .gitlab/sonar.yml:1]: Why are CI/CD files being modified?

File: host/src/include_private/nkv_framework.h [hunk 3]
```diff
@@ -229,8 +229,12 @@
     int32_t core_to_pin;
     int32_t path_numa_node;
     std::atomic<uint32_t> nkv_async_path_cur_qd;
-    std::unordered_map<std::size_t, std::set<std::string> > *listing_keys; 
-    std::unordered_map<std::string, nkv_value_wrapper*> *data_cache; 
+    // This is a vector of maps with keys and set values
+    std::vector<std::unordered_map<std::size_t, std::set<std::string>>> listing_keys;
+    //std::unordered_map<std::size_t, std::set<std::string> > *listing_keys;
```
Comment (benixon) [inline host/src/include_private/nkv_framework.h]: Remove commented code if verified.

File: host/src/include_private/nkv_framework.h [hunk 4]
```diff
@@ -229,8 +229,12 @@
...
     int32_t path_numa_node;
     std::atomic<uint32_t> nkv_async_path_cur_qd;
-    std::unordered_map<std::size_t, std::set<std::string> > *listing_keys; 
-    std::unordered_map<std::string, nkv_value_wrapper*> *data_cache; 
+    // This is a vector of maps with keys and set values
+    std::vector<std::unordered_map<std::size_t, std::set<std::string>>> listing_keys;
+    //std::unordered_map<std::size_t, std::set<std::string> > *listing_keys; 
+    // This is a vector of maps with keys and wrapper object values
+    std::vector<std::unordered_map<std::string, nkv_value_wrapper*>> data_cache;
+    //std::unordered_map<std::string, nkv_value_wrapper*> *data_cache;
```
Comment (benixon) [inline host/src/include_private/nkv_framework.h]: Remove commented code if verified.

File: host/src/include_private/nkv_framework.h [hunk 5]
```diff
@@ -282,16 +286,28 @@
         pthread_rwlock_init(&data_rw_lock_list[iter], NULL);
       }
 
-      listing_keys = new std::unordered_map<std::size_t, std::set<std::string> > (nkv_listing_cache_num_shards);
+      // Resize and initialize std::unordered_map at each index
+      listing_keys.resize(nkv_listing_cache_num_shards);
+      //listing_keys = new std::unordered_map<std::size_t, std::set<std::string> > (nkv_listing_cache_num_shards);
```
Comment (benixon) [inline host/src/include_private/nkv_framework.h]: Remove commented code if verified.

File: host/src/include_private/nkv_framework.h [hunk 6]
```diff
@@ -282,16 +286,28 @@
...
+      for (auto i =0; i<nkv_listing_cache_num_shards; i++) {
+          // Initialize map at index
+          listing_keys[i] = std::unordered_map<std::size_t, std::set<std::string>>();
+      }
       if (nkv_in_memory_exec) {
-        data_cache = new std::unordered_map<std::string, nkv_value_wrapper*> (nkv_listing_cache_num_shards);
+        //data_cache = new std::unordered_map<std::string, nkv_value_wrapper*> (nkv_listing_cache_num_shards);
```
Comment (benixon) [inline host/src/include_private/nkv_framework.h]: Remove commented code if verified.

File: host/src/include_private/nkv_framework.h (hunk 1 above)
Comment (ramachaitanyak) [reply host/src/include_private/nkv_framework.h:297]: Fixed

File: host/src/nkv_framework.cpp (hunk 2 above)
Comment (ramachaitanyak) [reply host/src/nkv_framework.cpp]: Fixed or updated appropriately

File: host/src/include_private/nkv_framework.h (hunk 6 above)
Comment (ramachaitanyak) [reply host/src/include_private/nkv_framework.h]: Fixed

File: host/src/include_private/nkv_framework.h (hunk 5 above)
Comment (ramachaitanyak) [reply host/src/include_private/nkv_framework.h]: Fixed

File: host/src/include_private/nkv_framework.h (hunk 4 above)
Comment (ramachaitanyak) [reply host/src/include_private/nkv_framework.h]: Fixed

File: host/src/include_private/nkv_framework.h (hunk 3 above)
Comment (ramachaitanyak) [reply host/src/include_private/nkv_framework.h]: Fixed

**Review by somnathr — APPROVED** [review]
Looks good

**Pull request merged**